pkg/upgrade/upgrader/jiva_upgrade.go

```diff
@@ -470,38 +444,52 @@ func (j *jivaVolumeOptions) replicaUpgrade(openebsNamespace string) error {
 
 	statusObj.Phase = utask.StepErrored
 
-	// Scaling down controller deployment before patching replica deployment
-	// if the replica is not upgraded already.
-	if j.replicaObj.version == currentVersion {
-		err = scaleTargetDeploy(j.controllerObj.name, j.ns, 0)
+	if currentVersion < "1.9.0" {
+		err = j.migrateReplica(openebsNamespace)
 		if err != nil {
-			statusObj.Message = "failed to scale down target depoyment"
+			statusObj.Message = "failed to migrate replica to openebs namespace"
 			statusObj.Reason = strings.Replace(err.Error(), ":", "", -1)
 			j.utaskObj, uerr = updateUpgradeDetailedStatus(j.utaskObj, statusObj, openebsNamespace)
 			if uerr != nil && isENVPresent {
 				return uerr
 			}
-			return errors.Wrap(err, "failed to scale down target depoyment")
+			return errors.Wrap(err, "failed to migrate replica to openebs namespace")
 		}
-	}
+	} else {
 
-	// replica patch
-	err = patchReplica(j.replicaObj, j.ns)
-	if err != nil {
-		// If patching of replica fails the controller needs to be reverted
-		// as the upgrade will not proceed with controller patch.
-		scaleErr := scaleTargetDeploy(j.controllerObj.name, j.ns, 1)
-		if scaleErr != nil {
-			klog.Infof("failed to scale up controller delpoyment. Please scale up deployment "+
-				"%s in %s namespace to 1 manually.", j.controllerObj.name, j.ns)
-		}
-		statusObj.Message = "failed to patch replica depoyment"
-		statusObj.Reason = strings.Replace(err.Error(), ":", "", -1)
-		j.utaskObj, uerr = updateUpgradeDetailedStatus(j.utaskObj, statusObj, openebsNamespace)
-		if uerr != nil && isENVPresent {
-			return uerr
+		// Scaling down controller deployment before patching replica deployment
+		// if the replica is not upgraded already.
+		if j.replicaObj.version == currentVersion {
+			err = scaleDeploy(j.controllerObj.name, j.ns, ctrlDeployLabel, 0)
```

Comment:
can this be moved to part of preUpgarde rather than being part of replicaUpgrade?